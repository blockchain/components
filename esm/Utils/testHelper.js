import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}