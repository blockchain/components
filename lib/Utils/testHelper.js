"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}