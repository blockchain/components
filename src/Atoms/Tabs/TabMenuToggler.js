// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled, { type ReactComponentFunctional } from 'styled-components'

import { FasBars, type IconPropsType } from '../../Atoms/Icons'
import { screenSize } from '../../Tools/interpolation'

type PropsType = {
  +children?: React.Node,
  +onToggle?: (SyntheticEvent<HTMLDivElement>) => void,
}

const Toggler: ReactComponentFunctional<IconPropsType> = styled(FasBars).attrs({
  selectable: true,
  size: '20px',
})`
  position: absolute;
  right: 20px;

  @media (min-width: ${screenSize('sm')}) {
    display: none;
  }
`

const TabMenuToggler = ({ onToggle, ...rest }: PropsType) => (
  <Toggler onClick={onToggle} {...rest} />
)

TabMenuToggler.propTypes = {
  children: PropTypes.node,
  onToggle: PropTypes.func,
}

export default TabMenuToggler
