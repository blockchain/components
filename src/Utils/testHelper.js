import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
