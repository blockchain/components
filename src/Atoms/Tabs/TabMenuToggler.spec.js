import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import TabMenuToggler from './TabMenuToggler'

describe('TabMenuToggler', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TabMenuToggler />)

    expect(tree.find(TabMenuToggler)).toMatchSnapshot()
  })
})
