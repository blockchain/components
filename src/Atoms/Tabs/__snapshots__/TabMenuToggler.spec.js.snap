// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenuToggler should render correctly 1`] = `
.c1 {
  fill: #545456;
  cursor: pointer;
}

.c1:hover {
  fill: #757679;
}

.c0 {
  position: absolute;
  right: 20px;
}

@media (min-width:48rem) {
  .c0 {
    display: none;
  }
}

<TabMenuToggler>
  <Styled(FasBars)>
    <FasBars
      className="c0"
      selectable={true}
      size="20px"
    >
      <Styled(SvgFasBars)
        className="c0"
        height="20px"
        selectable={1}
        width="20px"
      >
        <SvgFasBars
          className="c0 c1"
          height="20px"
          selectable={1}
          width="20px"
        >
          <svg
            className="c0 c1"
            height="20px"
            selectable={1}
            viewBox="0 0 448 512"
            width="20px"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </SvgFasBars>
      </Styled(SvgFasBars)>
    </FasBars>
  </Styled(FasBars)>
</TabMenuToggler>
`;
